pydicom/jsonrep.py

```diff
@@ -41,3 +47,202 @@ def convert_to_python_number(value, vr):
+            Key of the data element that contains the value
+            (options: ``{"Value", "InlineBinary", "BulkDataURI"}``)
+        bulk_data_uri_handler: callable or None
+            Callable function that accepts the "BulkDataURI" of the JSON
+            representation of a data element and returns the actual value of
+            that data element (retrieved via DICOMweb WADO-RS)
+        """
+        self.dataset_class = dataset_class
+        self.tag = tag
+        self.vr = vr
+        self.value = value
+        self.value_key = value_key
+        self.bulk_data_uri_handler = bulk_data_uri_handler
+
+    def get_element_values(self):
+        """Return a the data element value or list of values.
+
+        Returns
+        -------
+        str or bytes or int or float or dataset_class
+        or PersonName3 or PersonNameUnicode or list of any of these types
+            The value or value list of the newly created data element.
+        """
+        if self.value_key == 'Value':
+            if not isinstance(self.value, list):
+                fmt = '"{}" of data element "{}" must be a list.'
+                raise TypeError(fmt.format(self.value_key, self.tag))
+            if not self.value:
+                return empty_value_for_VR(self.vr)
+            element_value = []
+            for v in self.value:
+                element_value.append(self.get_regular_element_value(v))
+            if len(element_value) == 1 and self.vr != 'SQ':
+                element_value = element_value[0]
+            return convert_to_python_number(element_value, self.vr)
+
+        # The value for "InlineBinary" shall be encoded as a base64 encoded
+        # string, as shown in PS3.18, Table F.3.1-1, but the example in
+        # PS3.18, Annex F.4 shows the string enclosed in a list.
+        # We support both variants, as the standard is ambiguous here,
+        # and do the same for "BulkDataURI".
+        value = self.value
+        if isinstance(value, list):
+            value = value[0]
+
+        if self.value_key == 'InlineBinary':
+            if not isinstance(value, compat.char_types):
+                fmt = '"{}" of data element "{}" must be a bytes-like object.'
+                raise TypeError(fmt.format(self.value_key, self.tag))
+            return base64.b64decode(value)
+
+        if self.value_key == 'BulkDataURI':
+            if not isinstance(value, compat.string_types):
+                fmt = '"{}" of data element "{}" must be a string.'
+                raise TypeError(fmt.format(self.value_key, self.tag))
+            if self.bulk_data_uri_handler is None:
+                warnings.warn(
+                    'no bulk data URI handler provided for retrieval '
+                    'of value of data element "{}"'.format(self.tag)
+                )
+                return empty_value_for_VR(self.vr, raw=True)
+            return self.bulk_data_uri_handler(value)
+        return empty_value_for_VR(self.vr)
```

Comment:
When is it possible to get to this line? Aren't the only `value_key` options Value, InlineBinary and BulkDataURI?